### PR TITLE
ARROW-9602: [R] Improve cmake detection in Linux build

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: r/tools/linuxlibs.R [hunk 1]
```diff
@@ -315,8 +315,13 @@ build_libarrow <- function(src_dir, dst_dir) {
 }
 
 ensure_cmake <- function() {
-  cmake <- Sys.which("cmake")
-  if (!nzchar(cmake) || cmake_version() < 3.2) {
+  cmake <- find_cmake(c(
+    Sys.getenv("CMAKE"),
+    Sys.which("cmake"),
+    Sys.which("cmake3")
```
Comment (pitrou) [inline r/tools/linuxlibs.R:321]: Does `cmake3` actually exist in the wild?

File: r/tools/linuxlibs.R [hunk 2]
```diff
@@ -315,8 +315,13 @@ build_libarrow <- function(src_dir, dst_dir) {
...
+  ))
+
+  if (is.null(cmake)) {
     # If not found, download it
     cat("**** cmake\n")
     CMAKE_VERSION <- Sys.getenv("CMAKE_VERSION", "3.16.2")
```
Comment (pitrou) [inline r/tools/linuxlibs.R:327]: You probably want to bump that version: latest is either 3.18.1, 3.17.4 or 3.16.8.

**Review by pitrou — APPROVED** [review]
LGTM

File: r/tools/linuxlibs.R (hunk 1 above)
Comment (nealrichardson) [reply r/tools/linuxlibs.R:321]: Apparently so; it was in the report on https://issues.apache.org/jira/browse/ARROW-9602

File: r/tools/linuxlibs.R (hunk 2 above)
Comment (nealrichardson) [reply r/tools/linuxlibs.R:327]: Right, will do. Can't remember where I got that version number from before--probably one of our CI jobs.

**Pull request closed without merging**